tardis/visualization/plot_util.py

```diff
@@ -60,3 +73,255 @@ def get_mid_point_idx(arr):
+        """
+        Parse user requested species list and create list of species ids to be used.
+
+        Parameters
+        ----------
+        species_list : list of species to plot
+            List of species (e.g. Si II, Ca II, etc.) that the user wants to show as unique colours.
+            Species can be given as an ion (e.g. Si II), an element (e.g. Si), a range of ions
+            (e.g. Si I - V), or any combination of these (e.g. species_list = [Si II, Fe I-V, Ca])
+
+        """
+
+        if species_list is not None:
+            # check if there are any digits in the species list. If there are, then exit.
+            # species_list should only contain species in the Roman numeral
+            # format, e.g. Si II, and each ion must contain a space
+            if any(char.isdigit() for char in " ".join(species_list)) == True:
+                raise ValueError(
+                    "All species must be in Roman numeral form, e.g. Si II"
+                )
+            else:
+                full_species_list = []
+                for species in species_list:
+                    # check if a hyphen is present. If it is, then it indicates a
+                    # range of ions. Add each ion in that range to the list as a new entry
+                    if "-" in species:
+                        # split the string on spaces. First thing in the list is then the element
+                        element = species.split(" ")[0]
+                        # Next thing is the ion range
+                        # convert the requested ions into numerals
+                        first_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[0]
+                        )
+                        second_ion_numeral = roman_to_int(
+                            species.split(" ")[-1].split("-")[-1]
+                        )
+                        # add each ion between the two requested into the species list
+                        for ion_number in np.arange(
+                            first_ion_numeral, second_ion_numeral + 1
+                        ):
+                            full_species_list.append(
+                                f"{element} {int_to_roman(ion_number)}"
+                            )
+                    else:
+                        # Otherwise it's either an element or ion so just add to the list
+                        full_species_list.append(species)
+
+                # full_species_list is now a list containing each individual species requested
+                # e.g. it parses species_list = [Si I - V] into species_list = [Si I, Si II, Si III, Si IV, Si V]
+                self._full_species_list = full_species_list
+                requested_species_ids = []
+                keep_colour = []
+
+                # go through each of the requested species. Check whether it is
+                # an element or ion (ions have spaces). If it is an element,
```

Comment:
this is a dangerous assumption IMHO

Reply:
This should be fine as it is well documented that if the user doesn't include a space, the code will fall over. And it was the easiest way we could think of at the time for differentiating between a mixed list of ions and elements in a list.